soc/xtensa/intel_adsp/common/CMakeLists.txt

```diff
@@ -111,3 +113,61 @@ add_custom_command(
             $<TARGET_PROPERTY:bintools,strip_flag_final>
 )
 endif()
+
+find_program(RIMAGE NAMES rimage)
+
+if (DEFINED RIMAGE)
+message(STATUS "Found rimage ${RIMAGE}")
+else()
+message(STATUS "No rimage found")
+endif()
+
+# A direct env var RIMAGE_KEY_FILE may be given, otherwise
+# the board may define the RIMAGE_SIGN_KEY name along with an env
+# var to define the path to where key files are stored.
+if (DEFINED ENV{RIMAGE_KEY_FILE}
```

Comment:
That's exactly what happens, the boards define them for the open keys and config files. I wanted to enable a way to override that when an explicit key and conf file env are set for builds not using the predefined otc key and conf.

Reply:
> ... to override that when an explicit key and conf file env are set for builds not using the predefined otc key and conf.

This ability to override the default key without making git "dirty" is very important. Not just to sign official releases with a super top secret key but also because the only firmware difference between some products is the key, example:
- https://github.com/thesofproject/sof/pull/6070

Configuration file(s), environment variable(s), other... I don't know what is the best way to implement such an override, especially not when just coming back from vacation. I would instinctively prefer a configuration file not in git.

Reply:
There's a few env vars that can be used to override the defaults.